lib/jekyll/readers/data_reader.rb

```diff
@@ -4,11 +4,12 @@ module Jekyll
   class DataReader
     attr_reader :site, :content
 
-    def initialize(site)
+    def initialize(site, in_source_dir: nil)
       @site = site
       @content = {}
       @entry_filter = EntryFilter.new(site)
-      @source_dir = site.in_source_dir("/")
+      @in_source_dir = in_source_dir || @site.method(:in_source_dir)
+      @source_dir = @in_source_dir.call("/") if @in_source_dir
```

Comment:
Where is this variable `@source_dir` used?

Reply:
In line 58 where a debug statement states the location of the read data file. Not sure if we can drop this.

Reply:
I see..
The variable is only used to get a *relative path* of the data file.
That said, we can't remove it or modify it to maintain compatibility with any plugins subclassing the DataReader.

So, best leave it as is in current state.

Reply:
Okay, then we’re done here…